site/migration/01-migration.md

```diff
@@ -0,0 +1,257 @@
+`config.kubernetes.io/function` annotation. Functions can be declared in the
+pipeline section of Kptfile and invoked using `kpt fn render`. The declared function
+can point to the function config using `configPath` argument.
+
+#### Function Results
+
+In both `render` and `eval`, structured results can be enabled using the `--results-dir` flag.
+Please refer to the [function results] section for more information on the new structure of function results.
+
+### Changes to Setters
+
+`Setters` and `substitutions` no longer follow the complex and verbose OpenAPI format.
+The setters and substitutions are simplified to use new and simple setter pattern
+comments. Creating a setter is as simple as adding a line comment to the desired field.
+Users need not add any `OpenAPI` definitions. Please refer to [apply-setters] for
+the information about new syntax for setters and how to apply setter values, this is
+equivalent to the `kpt cfg set` in `v0.39.0`.
+[Setter Inheritance] works as usual in kpt `v1.0`.
+
+Here is the [simple example] of setter syntax transformation.
+
+#### Setter validations deprecated
+
+We want to keep the authoring experience of setters as simple as possible and
+hence [OpenAPI validations] and [required setters] feature offered by `v0.39.0` setters
+is no longer offered in `v1.0` version of kpt. However, we are working on providing
+an easy way to achieve similar functionality retaining the simplicity of setters,
+which is scoped for post `v1.0` release. Stay tuned.
+
+#### Auto-setters deprecated
+
+[Auto-setters] feature is deprecated in `v1.0` version of kpt. Since the setters are
+migrated to a new and simple declarative version, package consumers can easily
+declare all the setter values and render them all at once.
+
+### Changes to Live
+
+Kpt live in `v1.0` no longer uses an inventory object to track the grouping of resources
+in the cluster. Instead, it uses a more expressive `ResourceGroup` CRD. Please refer
+to the detailed guide on [migrating inventory objects] to the `ResourceGroup` equivalent.
+
+## Migration Steps
+
+Based on the changes discussed above, this section walks you through an end to end
+workflow to migrate an [example kpt package] which is compatible with `v0.39.0` version
+of kpt, and make it compatible with kpt `v1.0`.
+
+### Automated portion of migration
+
+Invoke `gcr.io/kpt-fn/fix` function from the root directory of the target kpt package.
+
+```shell
+# you must be using v1.0 version of kpt
+$ kpt fn eval --image gcr.io/kpt-fn/fix:unstable --include-meta-resources
+```
+
+#### Changes made by the function
+
+1. Best effort is made by the function to transform the `packageMetaData` section to
+   the `info` section.
+2. `upstream` section, in the `v1alpha1` Kptfile is converted to `upstream` and `upstreamLock`
+   sections in `v1alpha2` version of Kptfile.
+3. `dependencies` section is removed from Kptfile.
+4. Starlark functions section is removed from Kptfile.
```

Comment:
do you add the starlark function instead?

Reply:
No. Added it to manual portion.